### PR TITLE
Update runbooks to mention possibility to investigate memberlist KV store in various alerts

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

This PR updates Mimir runbooks to mention memberlist as possible cause of various alerts.

Fixes https://github.com/grafana/mimir/issues/1964

#### Checklist

- [na] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/operators-guide/mimir-runbooks/_index.md [hunk 1]
```diff
@@ -265,6 +265,7 @@ How to **investigate**:
 - If the failing service is going OOM (`OOMKilled`): scale up or increase the memory
 - If the failing service is crashing / panicking: look for the stack trace in the logs and investigate from there
   - If crashing service is query-frontend, querier or store-gateway, and you have "activity tracker" feature enabled, look for `found unfinished activities from previous run` message and subsequent `activity` messages in the log file to see which queries caused the crash.
+- When using Memberlist as KV store for hash rings, ensure that Memberlist is working correctly. See [instructions for `MimirGossipMembersMismatch` alert.](#MimirGossipMembersMismatch)
```
Comment (pracucci) [inline docs/sources/operators-guide/mimir-runbooks/_index.md:268]: I wouldn't link the ending dot. Also i think @grafana/docs-squad wants a link to match the destination text, so only link `MimirGossipMembersMismatch` (at least that's what I understood...):

```suggestion
- When using Memberlist as KV store for hash rings, ensure that Memberlist is working correctly. See instructions for [`MimirGossipMembersMismatch`](#MimirGossipMembersMismatch) alert.
```

_Same comment applies to other places._

File: docs/sources/operators-guide/mimir-runbooks/_index.md [hunk 2]
```diff
@@ -933,6 +940,8 @@ How to **investigate**:
 - Ensure Consul/Etcd is up and running.
 - Investigate the logs of the affected instance to find the specific error occurring when talking to Consul/Etcd.
 
+When using Memberlist as KV store for hash rings, all read and update operations work on a local copy of the hash ring, and will never fail and raise this alert.
```
Comment (pracucci) [inline docs/sources/operators-guide/mimir-runbooks/_index.md:943]: I would move this above, right after:

```
This alert fires if a Mimir instance is failing to run any operation on a KV store (eg. consul or etcd).
```

File: docs/sources/operators-guide/mimir-runbooks/_index.md [hunk 3]
```diff
@@ -831,12 +836,14 @@ This alert is fired when the multi-tenant alertmanager has been unable to check
 
 When the alertmanager loads its configuration on start up, when it polls for config changes or when there is a ring change it must check the ring to see if the tenant is still owned on this shard. To prevent one error from causing the loading of all configurations to fail we assume that on error the tenant is NOT owned for this shard. If checking the ring continues to fail then some tenants might not be assigned an alertmanager and might not be able to receive notifications for their alerts.
 
-The metric for this alert is cortex_alertmanager_ring_check_errors_total.
+The metric for this alert is `cortex_alertmanager_ring_check_errors_total`.
 
 How to **investigate**:
 
 Look at the error message that is logged and attempt to understand what is causing the failure. In most cases the error will be encountered when attempting to read from the ring, which can fail if there is an issue with in-use backend implementation.
```
Comment (pracucci) [inline docs/sources/operators-guide/mimir-runbooks/_index.md:843]: We can convert to an unordered list, like in other runbooks, now that we have 2 items.

**Review by pracucci — APPROVED** [review]

**Review by jhesketh — COMMENTED** [review]
lgtm;
Text in general looks fine. Not +A'ing due to other reviewers comments on formatting.

File: docs/sources/operators-guide/mimir-runbooks/_index.md (hunk 1 above)
Comment (pstibrany) [reply docs/sources/operators-guide/mimir-runbooks/_index.md:268]: Applied everywhere.

**Comment by pstibrany** [comment]
Thanks for reviews and suggestions. I applied them all. I will wait for review from @grafana/docs-squad before merging.

**Pull request merged**